### PR TITLE
fix(assemblyai): validate live options and preserve Unicode context

### PR DESCRIPTION
**Pull request by meeranmalik**
## Description

Focused fixes for the AssemblyAI option handling merged in #2409. This PR targets `main` and replaces #2458, which closed when its base branch was deleted after #2409 merged. The four fixes were not included in that merge; this is the same focused contribution reapplied to current main, with validation rerun.

## Changes Made

- Reuse a single Pro-family validation gate at construction and on both public update paths. Unsupported partial-transcript controls fail before configuration or keyterms are mutated or a provider message is queued.
- Reject model changes through `updateOptions` before mutating state. `speech_model` is not an [AssemblyAI UpdateConfiguration field](https://assemblyai.github.io/assemblyai-node-sdk/types/StreamingUpdateConfiguration.html); merely validating against the requested model would still send Pro-only options to the old server-side model. Create a new STT instance to change models. Repeating the current model, its equivalent alias, or an undefined model remains safe.
- Use the framework's `getBaseLanguage` helper so regional ISO-639-3 tags normalize and deduplicate correctly (for example, `eng-US` -> `en`, `cmn-Hans-CN` -> `zh`).
- Validate and truncate agent context using Unicode code points, matching the Python implementation and avoiding split surrogate pairs.
- Honor automatic context carryover opt-out and model support even when a fallback adapter forwards conversation items to every child. Explicit context updates on supported models remain allowed. Adds three regressions that failed before the guard, including an unsupported-model exception in the fallback path.
- Add mock-WebSocket regression tests for both update APIs, model/state preservation, language normalization, Unicode boundaries, and explicit false/zero values.

The context-carryover default introduced in #2409 is unchanged. No core interruption or endpointing policy is changed.

## Pre-Review Checklist

- [x] Build passes locally: `pnpm build` (40 tasks).
- [x] AI-generated code reviewed; deslop pass completed.
- [x] Changes and the explicit model-update rejection are explained above and in the changeset.
- [x] Scope limited to AssemblyAI option handling, its API report, tests, and changeset.
- [x] No file crosses 1000 lines; production `stt.ts` changes from 802 to 807 lines.
- [x] Shared validation replaces duplicated checks; existing language utility reused.
- [ ] Video demo / Agent Playground: not run; this is a provider-configuration fix tested with local WebSocket servers.

## Testing

- `ASSEMBLYAI_API_KEY= pnpm exec vitest run plugins/assemblyai agents/src/language.test.ts agents/src/stt/fallback_adapter.test.ts`: 76 passed, 1 credential-dependent skip.
- `pnpm build`: passed.
- `pnpm -w format:write`: completed. Repository-wide `pnpm -w lint:fix` is blocked by an unchanged `@typescript-eslint/no-misused-promises` error in `plugins/openai/src/ws/llm.ts:127:50` on current main. An unrelated Cartesia import autofix was excluded from this PR.
- Focused Prettier check, AssemblyAI lint, and `api:check`: passed.
- `git diff --check`: passed.

The local review first reproduced six failing assertions across the four issues; the regression suite now passes. No credentialed AssemblyAI integration or real audio call was run. The full monorepo test suite was not run.

### CODE REVIEW

File: plugins/assemblyai/src/stt.ts [hunk 1]
```diff
@@ -324,7 +335,9 @@ export class STT extends stt.STT {
   override _pushConversationItem(ev: ConversationItemAddedEvent): void {
     const chatItem = ev.item;
     if (chatItem instanceof ChatMessage && chatItem.role === 'assistant' && chatItem.textContent) {
-      this.updateOptions({ agentContext: chatItem.textContent.slice(-MAX_AGENT_CONTEXT_CHARS) });
+      this.updateOptions({
+        agentContext: Array.from(chatItem.textContent).slice(-MAX_AGENT_CONTEXT_CHARS).join(''),
+      });
```
[the remark this answers is not in the document]
Comment (meeranmalik) [reply plugins/assemblyai/src/stt.ts:341]: Fixed in b866c280. The AssemblyAI conversation-item hook now checks capabilities.chatContext before updating context, so fallback forwarding respects both explicit opt-out and model support. Added regressions using the real STT fallback adapter plus a local mock WebSocket: opted-out children receive no automatic context, opted-in children still receive it, explicit context updates remain allowed on supported models, and unsupported children do not throw. All three new cases failed before the guard; the combined AssemblyAI, language, and STT fallback suites now pass (76 passed, 1 credential-dependent skip). Full build and AssemblyAI lint/API checks also pass.

**Review by tinalenguyen — APPROVED** [review]
thanks for the PR!
